### PR TITLE
Add table aws_emr_instance_group. Closes #550, closes #551

### LINKED ISSUE #550: Add table aws_emr_instance_group
**References**
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/emr_instance_group

### LINKED ISSUE #551: Add a column to store `instance_type` in table `aws_emr_cluster`
**Is your feature request related to a problem? Please describe.**
Didn't find any column storing instance_type  for emr cluster

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Subhajit97**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT 300

SETUP: tests/aws_emr_instance_group []

PRETEST: tests/aws_emr_instance_group

TEST: tests/aws_emr_instance_group
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_key_pair.deployer: Creating...
aws_iam_role.iam_emr_profile_role: Creating...
aws_iam_role.iam_emr_service_role: Creating...
aws_vpc.my_vpc: Creating...
aws_key_pair.deployer: Creation complete after 2s [id=deployer-key]
aws_iam_role.iam_emr_service_role: Creation complete after 4s [id=turbottest91412_1]
aws_iam_role.iam_emr_profile_role: Creation complete after 4s [id=turbottest91412_2]
aws_iam_instance_profile.emr_profile: Creating...
aws_iam_role_policy.iam_emr_profile_policy: Creating...
aws_iam_role_policy.iam_emr_service_policy: Creating...
aws_iam_role_policy.iam_emr_profile_policy: Creation complete after 2s [id=turbottest91412_2:turbottest91412_2]
aws_iam_role_policy.iam_emr_service_policy: Creation complete after 3s [id=turbottest91412_1:turbottest91412_1]
aws_iam_instance_profile.emr_profile: Creation complete after 4s [id=var.resource_name]
aws_vpc.my_vpc: Still creating... [10s elapsed]
aws_vpc.my_vpc: Creation complete after 12s [id=vpc-026c710e2dbfe8bf2]
aws_internet_gateway.gw: Creating...
aws_subnet.my_subnet: Creating...
aws_subnet.my_subnet: Creation complete after 4s [id=subnet-0c7c4662cd8745a61]
aws_security_group.emr_master: Creating...
aws_internet_gateway.gw: Creation complete after 5s [id=igw-02e9d7e6f1857ddc6]
aws_route_table.test_route_table: Creating...
aws_route_table.test_route_table: Creation complete after 5s [id=rtb-0ea5b05f30bfc10a9]
aws_main_route_table_association.test_association: Creating...
aws_security_group.emr_master: Creation complete after 8s [id=sg-0c3680eefecda4221]
aws_emr_cluster.named_test_resource: Creating...
aws_main_route_table_association.test_association: Creation complete after 2s [id=rtbassoc-0487dc8ac1d98d596]
aws_emr_cluster.named_test_resource: Still creating... [10s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [20s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [30s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [40s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [50s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [1m0s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [1m10s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [1m20s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [1m30s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [1m40s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [1m50s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [2m0s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [2m10s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [2m20s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [2m30s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [2m40s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [2m50s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [3m0s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [3m10s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [3m20s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [3m30s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [3m40s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [3m50s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [4m0s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [4m10s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [4m20s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [4m30s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [4m40s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [4m50s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [5m0s elapsed]
aws_emr_cluster.named_test_resource: Still creating... [5m10s elapsed]
aws_emr_cluster.named_test_resource: Creation complete after 5m14s [id=j-1IPL37X41FTS0]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals

Apply complete! Resources: 13 added, 0 changed, 0 destroyed.

Outputs:

region_name = us-east-1
resource_id = j-1IPL37X41FTS0
resource_name = turbottest91412

Running SQL query: test-list-query.sql
[
  {
    "cluster_id": "j-1IPL37X41FTS0",
    "configurations_version": 0,
    "instance_group_type": "MASTER",
    "instance_type": "m4.large",
    "last_successfully_applied_configurations_version": 0,
    "market": "ON_DEMAND",
    "requested_instance_count": 1,
    "running_instance_count": 1
  }
]
✔ PASSED

POSTTEST: tests/aws_emr_instance_group

TEARDOWN: tests/aws_emr_instance_group

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
### Basic info

```sql
select
  id,
  arn,
  cluster_id,
  instance_group_type,
  state
from
  aws_emr_instance_group;
```
```
+------------------+--------------------------------------------------------------------+-----------------+---------------------+------------+
| id               | arn                                                                | cluster_id      | instance_group_type | state      |
+------------------+--------------------------------------------------------------------+-----------------+---------------------+------------+
| ig-2372S3K11PGF7 | arn:aws:emr:us-east-1:856090093175:instance-group/ig-2372S3K11PGF7 | j-1IPL37X41FTS0 | MASTER              | TERMINATED |
| ig-15LW0PRUB4B5O | arn:aws:emr:us-east-1:856090093175:instance-group/ig-15LW0PRUB4B5O | j-3VSS4KPKPCFGC | MASTER              | TERMINATED |
| ig-WVRIL2Q83ONR  | arn:aws:emr:us-east-1:856090093175:instance-group/ig-WVRIL2Q83ONR  | j-3VSS4KPKPCFGC | CORE                | TERMINATED |
| ig-2YBNOCDZ0CKLB | arn:aws:emr:us-east-1:856090093175:instance-group/ig-2YBNOCDZ0CKLB | j-1Q3PEYUEAM8FP | MASTER              | TERMINATED |
| ig-1AR33EU9Y4PPI | arn:aws:emr:us-east-1:856090093175:instance-group/ig-1AR33EU9Y4PPI | j-1Q3PEYUEAM8FP | CORE                | TERMINATED |
| ig-398ULP48WQXAF | arn:aws:emr:us-east-1:856090093175:instance-group/ig-398ULP48WQXAF | j-1BHOKWIEQKMSX | MASTER              | TERMINATED |
| ig-23TFWGQNTPR7Z | arn:aws:emr:us-east-1:856090093175:instance-group/ig-23TFWGQNTPR7Z | j-2YVSOXPKWZ52C | CORE                | TERMINATED |
| ig-2V80P3WL17KGY | arn:aws:emr:us-east-1:856090093175:instance-group/ig-2V80P3WL17KGY | j-2YVSOXPKWZ52C | MASTER              | TERMINATED |
+------------------+--------------------------------------------------------------------+-----------------+---------------------+------------
```

### Get the master instance type used for a cluster

```sql
select
  ig.id as instance_group_id,
  ig.cluster_id,
  c.name as cluster_name,
  ig.instance_type
from
  aws_emr_instance_group as ig,
  aws_emr_cluster as c
where
  ig.cluster_id = c.id
  and ig.instance_group_type = 'MASTER';
```
```
+-------------------+-----------------+-----------------+---------------+
| instance_group_id | cluster_id      | cluster_name    | instance_type |
+-------------------+-----------------+-----------------+---------------+
| ig-2372S3K11PGF7  | j-1IPL37X41FTS0 | turbottest91412 | m4.large      |
| ig-15LW0PRUB4B5O  | j-3VSS4KPKPCFGC | turbottest34420 | m4.large      |
| ig-398ULP48WQXAF  | j-1BHOKWIEQKMSX | turbottest33504 | m4.large      |
| ig-2YBNOCDZ0CKLB  | j-1Q3PEYUEAM8FP | turbottest97100 | m4.large      |
| ig-2V80P3WL17KGY  | j-2YVSOXPKWZ52C | turbottest86231 | m4.large      |
+-------------------+-----------------+-----------------+---------------+
```

### Get the count of running instances (core and master) per cluster

```sql
select
  cluster_id,
  sum(running_instance_count) as running_instance_count
from
  aws_emr_instance_group
where
  state = 'RUNNING'
group by cluster_id;
```
```
+-----------------+-----+
| cluster_id      | sum |
+-----------------+-----+
| j-1BHOKWIEQKMSX | 1   |
| j-1IPL37X41FTS0 | 1   |
| j-1Q3PEYUEAM8FP | 2   |
| j-2YVSOXPKWZ52C | 2   |
| j-3VSS4KPKPCFGC | 2   |
+-----------------+-----+
```
</details>

### CODE REVIEW

File: aws-test/tests/aws_emr_instance_group/variables.tf [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+      "Principal": {
+        "Service": "elasticmapreduce.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+resource "aws_iam_role_policy" "iam_emr_service_policy" {
```
Comment (bigdatasourav) [inline aws-test/tests/aws_emr_instance_group/variables.tf:67]:
```suggestion

resource "aws_iam_role_policy" "iam_emr_service_policy" {
```

File: aws-test/tests/aws_emr_instance_group/variables.tf [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+            "sqs:Delete*",
+            "sqs:GetQueue*",
+            "sqs:PurgeQueue",
+            "sqs:ReceiveMessage"
+        ]
+    }]
+}
+EOF
+}
+resource "aws_iam_role" "iam_emr_profile_role" {
```
Comment (bigdatasourav) [inline aws-test/tests/aws_emr_instance_group/variables.tf:135]:
```suggestion

resource "aws_iam_role" "iam_emr_profile_role" {
```

File: aws-test/tests/aws_emr_instance_group/variables.tf [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+resource "aws_iam_instance_profile" "emr_profile" {
```
Comment (bigdatasourav) [inline aws-test/tests/aws_emr_instance_group/variables.tf:153]:
```suggestion

resource "aws_iam_instance_profile" "emr_profile" {
```

File: aws-test/tests/aws_emr_instance_group/variables.tf [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+  name = "var.resource_name"
+  role = aws_iam_role.iam_emr_profile_role.name
+}
+resource "aws_iam_role_policy" "iam_emr_profile_policy" {
```
Comment (bigdatasourav) [inline aws-test/tests/aws_emr_instance_group/variables.tf:157]:
```suggestion

resource "aws_iam_role_policy" "iam_emr_profile_policy" {
```

File: aws-test/tests/aws_emr_instance_group/variables.tf [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+            "s3:*",
+            "sdb:*",
+            "sns:*",
+            "sqs:*"
+        ]
+    }]
+}
+EOF
+}
+resource "aws_vpc" "my_vpc" {
```
Comment (bigdatasourav) [inline aws-test/tests/aws_emr_instance_group/variables.tf:184]:
```suggestion

resource "aws_vpc" "my_vpc" {
```

File: aws-test/tests/aws_emr_instance_group/variables.tf [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+  cidr_block = "168.31.0.0/16"
+}
+resource "aws_security_group" "emr_master" {
```
Comment (bigdatasourav) [inline aws-test/tests/aws_emr_instance_group/variables.tf:187]:
```suggestion

resource "aws_security_group" "emr_master" {
```

File: aws/table_aws_emr_instance_group.go [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+type instanceGroupDetails = struct {
+	emr.InstanceGroup
+	ClusterID string
+}
+
+//// LIST FUNCTION
+
+func listEmrInstanceGroups(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var region string
```
Comment (bigdatasourav) [inline aws/table_aws_emr_instance_group.go:164]: No need to pass region.

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**